frmts/netcdf/netcdfdrivercore.cpp

```diff
@@ -250,9 +250,11 @@ struct NCDFDriverSubdatasetInfo : public GDALSubdatasetInfo
             m_driverPrefixComponent = aosParts[0];
 
             int subdatasetIndex{2};
+
             const bool hasDriveLetter{
-                (strlen(aosParts[1]) == 2 && std::isalpha(aosParts[1][1])) ||
-                (strlen(aosParts[1]) == 1 && std::isalpha(aosParts[1][0]))};
+                (strlen(aosParts[2]) > 1 && aosParts[2][0] == '\\') &&
```

Comment:
actually my guess is that msys2 uses forward slash, not backward slash. We should handle both here an in HDF4/HDF5

Reply:
> I'm afraid we can't: what is C in this case? "NETCDF:C:/navigation_data/longitude"

the Open() code makes the guess this is a letter driver.

> I don't know how to fix it. Maybe some windows-specific code?

have a look at the logic in netcdfdataset.cpp lines 8983-9001. Not pretty, but works in practice
(my initial idea was that the GetSubdatasetInfo() would 'blindly' port the existing parsing logic, and that Open() would call GetSubdatasetInfo() to avoid duplication of code)